Sources/Player/PlaybackEngine/Internal/AVQueuePlayer/Legacy/AVQueuePlayerWrapperLegacy.swift

```diff
@@ -154,9 +159,26 @@ final class AVQueuePlayerWrapperLegacy: GenericMediaPlayer {
 
 			self.delegates.seeking(in: asset)
 
-			let completed = await self.player.seek(to: time)
-			guard completed, currentItem == self.player.currentItem, self.player.timeControlStatus == .playing else {
-				return
+			let seekTime = CMTime(seconds: time, preferredTimescale: 1000)
+			if self.shouldPauseAndPlayAroundSeek {
+				if self.player.timeControlStatus == .playing {
+					self.isSeeking = true
+					await self.player.pause()
+				}
+
+				let completed = await self.player.seek(to: time)
+				self.isSeeking = false
+				await self.player.play()
```

Comment:
if seek failed and completed is false, should we do play here?

Reply:
Also, later we do 
```
self.delegates.playing(asset: asset)
```

But I think that should not be done in the case where we manually call play (It will do that already, so now we repeat the delegate callback)

Reply:
(This applies to both Player implementations)

Reply:
Yes, I was not sure about the failed scenario. Let me do some testing around the unhappy scenario.

Reply:
I tried to test to get actual completed false - I could just force it, but this doesn't apply to the actual player returning false since I would guess the internal state would be different.
I think if it fails/completed is false, we should still try to play it. This I believe would have the same behavior as current scenario when we don’t do pause/play, so it would follow the same existing approach.
As for the `self.delegates.playing(asset: asset)`, indeed there's no need to call it. I did remove it at some point - just saw I removed in the non legacy version but missed adding the asset position, so in the end both needed adjustments after I rearranged the code after testing.

Reply:
removed duplicate playing call, and adjusted both players for the asset position.